File: contests/lc/2125.NumberofLaserBeamsinaBank/solution.go
package numberoflaserbeamsinabank

import (
	"strings"
)

func countOnes(s string) int {
	return strings.Count(s, "1")
}

func numberOfBeams(bank []string) int {
	totalBeams := 0
	prevCount := 0

	for _, row := range bank {
		currentCount := countOnes(row)
		if currentCount == 0 {
			continue
		}
		totalBeams += prevCount * currentCount
		prevCount = currentCount
	}

	return totalBeams
}

File: contests/lc/2125.NumberofLaserBeamsinaBank/solution.py
from typing import List

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        total_beams = 0
        prev_count = 0  # Количество устройств в последней непустой строке

        for row in bank:
            count = row.count('1')  # Считаем количество '1' в текущей строке
            if count == 0:
                continue  # Пропускаем строки без устройств
            total_beams += prev_count * count  # Количество лучей между строками
            prev_count = count  # Обновляем предыдущее количество

        return total_beams